controllers/storagecluster/provider_server.go

```diff
@@ -434,3 +444,62 @@ func RandomString(l int) string {
 
 	return string(bytes)
 }
+
+func getOnboardingJobObject(instance *ocsv1.StorageCluster) *batchv1.Job {
+
+	return &batchv1.Job{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      onboardingJobName,
+			Namespace: instance.Namespace,
+		},
+		Spec: batchv1.JobSpec{
+			Template: corev1.PodTemplateSpec{
+				Spec: corev1.PodSpec{
+					RestartPolicy:      corev1.RestartPolicyOnFailure,
+					ServiceAccountName: onboardingJobName,
+					Containers: []corev1.Container{
+						{
+							Name:    onboardingJobName,
+							Image:   os.Getenv(onboardingSecretGeneratorImage),
+							Command: []string{"/usr/local/bin/onboarding-secret-generator"},
+							Env: []corev1.EnvVar{
+								{
+									Name:  "OPERATOR_NAMESPACE",
+									Value: instance.Namespace,
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func (o *ocsProviderServer) createJob(r *StorageClusterReconciler, instance *ocsv1.StorageCluster) (reconcile.Result, error) {
+	if os.Getenv(onboardingSecretGeneratorImage) == "" {
+		r.Log.Info("onboardingSecretGeneratorImage env var is not set.")
+	} else {
+		onboardingSecretGeneratorJob := &batchv1.Job{}
+
+		err := r.Client.Get(context.TODO(), types.NamespacedName{Namespace: instance.Namespace, Name: onboardingJobName}, onboardingSecretGeneratorJob)
+
+		if err == nil {
+			for _, condition := range onboardingSecretGeneratorJob.Status.Conditions {
```

Comment:
this doesn't help in rotating the secrets, like if the job is already run and when the rest api deleted the secret, we need to create a job for recreating it even when the previous one was completed